.github/workflows/codeql-analysis.yml

```
@@ -7,55 +7,80 @@ on:
       # Step 7: Build iOS (single build)
       - name: Build Flutter iOS
+        id: build-ios
         run: flutter build ios --no-codesign
 
-      # Step 8: Perform the CodeQL analysis
+      # Step 8: Perform the CodeQL Analysis
       - name: Perform CodeQL Analysis
-        uses: github/codeql-action/analyze@v2
\ No newline at end of file
+        id: analyze
+        uses: github/codeql-action/analyze@v3
```

yamllint (1.35.1)

[error] no new line character at the end of file

(new-line-at-end-of-file)